File: ios/SwiftKotlination/SwiftKotlination/Sources/Network/NetworkError.swift
enum NetworkError: String, ErrorStringConvertible, Codable {
    case invalidData
    case invalidResponse
    case invalidRequest

    var description: String {
        switch self {
        case .invalidRequest:
            return "Invalid request, please try again later."
        case .invalidResponse:
            return "Invalid response, please try again later."
        case .invalidData:
            return "Invalid data, please try again later."
        }
    }
}

File: android/SwiftKotlination/app/src/main/kotlin/fr/jhandguy/swiftkotlination/network/NetworkError.kt
package fr.jhandguy.swiftkotlination.network

sealed class NetworkError: Error() {
    class InvalidRequest: NetworkError()
    class InvalidResponse: NetworkError()

    val description: String
        get() = when(this) {
            is InvalidRequest   -> "Invalid request, please try again later."
            is InvalidResponse  -> "Invalid response, please try again later."
        }
}